static/app/views/performance/transactionSummary/transactionSpans/spanDetails/exclusiveTimeHistogram.tsx

```diff
@@ -0,0 +1,240 @@
+          )}
+        />
+      </HeaderTitleLegend>
+      <ChartZoom
+        router={router}
+        period={period}
+        start={start}
+        end={end}
+        utc={utc === 'true'}
+      >
+        {zoomRenderProps => (
+          <SpanHistogramQuery
+            location={location}
+            orgSlug={organization.slug}
+            eventView={eventView}
+            numBuckets={NUM_BUCKETS}
+            precision={PRECISION}
+            span={spanSlug}
+            dataFilter="exclude_outliers"
+          >
+            {({histogram, isLoading, error}) => {
+              if (error) {
+                return (
+                  <ErrorPanel data-test-id="histogram-error-panel">
+                    <IconWarning color="gray300" size="lg" />
+                  </ErrorPanel>
+                );
+              }
+
+              const chartOptions = {
+                grid: {
+                  left: '10px',
+                  right: '10px',
+                  top: '40px',
+                  bottom: '0px',
+                },
+                colors: theme.charts.getColorPalette(yAxis.length - 2),
+                seriesOptions: {
+                  showSymbol: false,
+                },
+                tooltip: {
+                  trigger: 'axis' as const,
+                  // p50() coerces the axis to be time based
+                  valueFormatter: (value, _seriesName) =>
+                    tooltipFormatter(value, 'p50()'),
+                },
+                yAxis: {
+                  axisLabel: {
+                    color: theme.chartLabel,
+                    // p50() coerces the axis to be time based
+                    formatter: (value: number) => axisLabelFormatter(value, 'p50()'),
+                  },
+                },
+              };
+
+              return (
+                <TransitionChart loading={isLoading} reloading={isLoading}>
+                  <TransparentLoadingMask visible={isLoading} />
+                  <Chart
+                    {...zoomRenderProps}
+                    {...chartOptions}
+                    isLoading={isLoading}
+                    isErrored={!!error}
+                    chartData={histogram}
+                    location={location}
+                  />
+                </TransitionChart>
+              );
+            }}
+          </SpanHistogramQuery>
+        )}
+      </ChartZoom>
+    </Fragment>
+  );
+}
+
+type ChartProps = {
+  chartData: HistogramData | null;
+  isErrored: boolean;
+  isLoading: boolean;
+  location: Location;
+  colors?: string[];
+  disableChartPadding?: boolean;
+  disableXAxis?: boolean;
+  disableZoom?: boolean;
+  grid?: BarChart['props']['grid'];
+  height?: number;
+};
+
+export function Chart(props: ChartProps) {
+  const {
+    isLoading,
+    chartData,
+    location,
+    height,
+    grid,
+    disableXAxis,
+    disableZoom,
+    colors,
+  } = props;
+
+  if (!chartData) {
+    return null;
```

Comment:
Might be the source of the height dropping to 0. Considering add a placeholder / LoadingPanel in the loading state.